src/Nox.Ui.Blazor.Lib/Components/NoxTypes/ViewArea.razor.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.AspNetCore.Components;
+using Nox.Types;
+using Nox.Ui.Blazor.Lib.Models;
+
+namespace Nox.Ui.Blazor.Lib.Components.NoxTypes;
+
+public partial class ViewArea : ComponentBase
+{
+
+    #region Declarations
+
+    [Parameter]
+    public Decimal? Area { get; set; }
+
+    [Parameter]
+    public AreaUnit AreaUnit { get; set; } = AreaUnit.SquareMeter;
+
+    [Parameter]
+    public string Format { get; set; } = "#,##0.##";
+
+    public string DisplayArea
+    {
+
+        get
+        {
+            if (Area != null)
+            {
+                switch (Area)
+                {
+                    case < 0:
+                        float rtnMinusArea = 0;
+                        return rtnMinusArea.ToString(Format) + DisplayAreaUnit();
+                    default:
+                        return Area?.ToString(Format) + DisplayAreaUnit();
+                }
+            }
+
+            return String.Empty;
+        }
+    }
+
+    private string DisplayAreaUnit()
+    {
+        if (AreaUnit == AreaUnit.SquareFoot)
```

Comment:
we can use switch here:
```suggestion
        switch(AreaUnit)
        ...
```

Reply:
unfortunately couldnt get switch to work - as always says 'constant error' so left as if statements instead of doing something more complicated with nameof etc